### DISCUSSION TITLE
How do I set the color of an object?

### DISCUSSION THREAD

**Discussion by jbaicoianu** [jbaicoianu/janusweb#244] [FAQs]
How do I set the initial color of an object, and how do I update it from a script?

**Comment by lbgrid** [comment]
To set to initial color of an object, you can simply set the col parameter of the object in your JML:
```
<FireBoxRoom>
    <room>
        <object id="cube" js_id="color_changing_cube" col="1 1 1"></object>
    </room>
</FireBoxRoom>
```

You can then use the js_id of your object to update the object's color, or any other setting in your room script:
```
let n=50; // we can use this value to implement a timer to change the color
let old_n=n;
let color_changing_cube=room.objects['color_changing_cube'];
getRandomColor=function(){
    return Math.random()+' '+Math.random()+' '+Math.random();
}
room.onload=function(){
    color_changing_cube.col='0 0 0'; //this will change the cube frm it's initial color (white) to black when the room loads
}
room.update=function(dt){
    /* We can set up a timer to change the color every n cycles */ 
    n--;
    if(n<=1){
        n=old_n;
        color_changing_cube.col=getRandomColor();
    }
}
```
You could also use many other techniques to trigger the desired change in color value, such as a touch event, gaze event, proximity trigger, or even a 2D element with a onload or onclick event.
